File: src/main/java/com/suhailkandanur/eopi/util/Node.java
package com.suhailkandanur.eopi.util;

/**
 * Created by suhail on 2016-10-30.
 */
public class Node {
    private int data;
    private Node next;

    public Node(int data) {
        this.data = data;
        next = null;
    }

    public int getData() {
        return data;
    }

    public Node getNext() {
        return next;
    }

    public void setData(int data) {
        this.data = data;
    }

    public void setNext(Node next) {
        this.next = next;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Node node = (Node) o;

        if (data != node.data) return false;
        return next != null ? next.equals(node.next) : node.next == null;
    }

    @Override
    public int hashCode() {
        int result = data;
        result = 31 * result + (next != null ? next.hashCode() : 0);
        return result;
    }
}

File: python/Node/Node.py
class Node:
    def __init__(self, data, next=None, prev=None):
        self.data = data
        self.next = next
        self.previous = prev

    def getNext(self):
        return self.next

    def getPrevious(self):
        return self.previous

    def getData(self):
        return self.data

    def setNext(self, next):
        self.next = next

    def setPrevious(self, prev):
        self.previous = prev

